core/modules/rproc/ProtoRowBuffer.cc

```diff
@@ -24,136 +24,69 @@
-          default: *destI++ = *i; break;
-            // Null (\N) is not treated by escaping in this context.
-        }
-    }
-    return destI - destBegin;
-}
+namespace {
 
-/// Copy a rawColumn to an STL container
-template <typename T>
-inline int copyColumn(T& dest, std::string const& rawColumn) {
-    int existingSize = dest.size();
-    dest.resize(existingSize + 2 + 2 * rawColumn.size());
-    dest[existingSize] = '\'';
-    int valSize = escapeString(dest.begin() + existingSize + 1,
-                               rawColumn.begin(), rawColumn.end());
-    dest[existingSize + 1 + valSize] = '\'';
-    dest.resize(existingSize + 2 + valSize);
-    return 2 + valSize;
-}
-////////////////////////////////////////////////////////////////////////
-// ProtoRowBuffer
-////////////////////////////////////////////////////////////////////////
+LOG_LOGGER _log = LOG_GET("lsst.qserv.rproc.ProtoRowBuffer");
 
-/// ProtoRowBuffer is an implementation of RowBuffer designed to allow a
-/// LocalInfile object to use a Protobufs Result message as a row source
-class ProtoRowBuffer : public mysql::RowBuffer {
-public:
-    ProtoRowBuffer(proto::Result& res);
-    virtual unsigned fetch(char* buffer, unsigned bufLen);
-
-private:
-    void _initCurrentRow();
-    void _initSchema();
-    void _readNextRow();
-    // Copy a row bundle into a destination STL char container
-    template <typename T>
-    int _copyRowBundle(T& dest, proto::RowBundle const& rb) {
-        int sizeBefore = dest.size();
-        for(int ci=0, ce=rb.column_size(); ci != ce; ++ci) {
-            if (ci != 0) {
-                dest.insert(dest.end(), _colSep.begin(), _colSep.end());
-            }
-
-            if (!rb.isnull(ci)) {
-                copyColumn(dest, rb.column(ci));
-            } else {
-                dest.insert(dest.end(), _nullToken.begin(), _nullToken.end() );
-            }
+// Print the contents of a char vector, using ascii values for non-printing characters.
+std::string printCharVect(std::vector<char> const& cVect) {
+    std::string str;
+    for (char c : cVect) {
+        if (std::isprint(c)) {
+            str += c;
+        } else {
+            str += std::string("~") + std::to_string(c) + "~";
         }
-        return dest.size() - sizeBefore;
     }
+    return str;
+}
+
+} // namespace
 
-    std::string _colSep; ///< Column separator
-    std::string _rowSep; ///< Row separator
-    std::string _nullToken; ///< Null indicator (e.g. \N)
-    proto::Result& _result; ///< Ref to Resultmessage
 
-    sql::Schema _schema; ///< Schema object
-    int _rowIdx; ///< Row index
-    int _rowTotal; ///< Total row count
-    std::vector<char> _currentRow; ///< char buffer representing current row.
-};
+namespace lsst {
+namespace qserv {
+namespace rproc {
 
-ProtoRowBuffer::ProtoRowBuffer(proto::Result& res)
+ProtoRowBuffer::ProtoRowBuffer(proto::Result& res, int jobId, std::string const& jobIdColName,
```

Comment:
LSST C++ Style Guide (section 6-6) requires the parameters to be aligned if possible.